sonar-helm-for-iac/src/converters/io-handling.go

```diff
@@ -98,17 +104,41 @@ func (s StdinReader) readInput(scanner *bufio.Scanner, nLines int) ([]byte, erro
 	linesToRead := nLines
 	for scanner.Scan() {
 		rawInput = append(rawInput, scanner.Bytes()...)
-		rawInput = append(rawInput, []byte("\n")...)
 		linesToRead--
 		if linesToRead == 0 {
 			break
 		}
 	}
-	// the last delimiter added needs to be removed again
-	if len(rawInput) > 0 {
-		rawInput = rawInput[:len(rawInput)-1]
-	}
 
 	// if scanner has encountered an error, scanner.Err will return it here
 	return rawInput, scanner.Err()
 }
+
+// By default bufio.Scanner scans input line by line using \n (LR) as separator, trims \r (CR)
+// and doesn't include new line separator in token.
+// This fuction split tokens using \r\n or \r or \n or \u2028 or \u2029 and include them in tokens.
+func ScanLinesIncludeNewLine(data []byte, atEOF bool) (advance int, token []byte, err error) {
+	if atEOF && len(data) == 0 {
+		return 0, nil, nil
+	}
+
+	indexCRLF := bytes.Index(data, []byte("\r\n"))
+	indexNewLine := bytes.IndexAny(data, "\r\n\u2028\u2029")
+
+	if indexCRLF >= 0 && indexCRLF <= indexNewLine {
+		// line ends with CRLR
+		return indexCRLF + 2, data[0 : indexCRLF+2], nil
```

Comment:
I don't quite understand. If your idea is to keep all separators in the resulting string, why are you only trimming `\n` in the `io-hanling.go`? Or is the idea that since we only append LF on the Java side, all other separators are valid tokens in the input text and because of that should be kept?

Reply:
Ok, and `template` can manage to parse the input with any separators, so afterward it works fine? And by keeping the original separators we get correct offsets with respect to the original input?

Reply:
Yes & yes